src/components/Editor/SearchBar.js

```diff
@@ -660,10 +664,16 @@ class SearchBar extends Component {
     const { symbolSearchOn, selectedSymbolType } = this.props;
 
     function searchTypeBtn(searchType) {
+      let active =
+        (symbolSearchOn &&
+          selectedSymbolType == searchType &&
+          searchType !== "text") ||
+        (!symbolSearchOn && searchType === "text");
```

Comment:
This is a pretty complex boolean. I think it would be more readable if we split this into functions. Maybe something like this:

```
const { symbolSearchOn, selectedSymbolType } = this.props;

function isSymbolSearchOn(searchType) {
  // since the boolean is so complex, consisting of three parts, lets break it up
  // and use clear variable names to make it obvious what is happening
  const isSymbolSearch = selectedSymbolType === searchType;
  const notSearchText = searchType !== "text";

  return (symbolSearchOn && isSearchType && notSearchText);
}

function isSearchText(searchType) {
  return (!symbolSearchOn && searchType === "text");
}

function isTextSearch(searchType) {
  const { symbolSearchOn } = this.props;
  return (symbolSearchOn && searchType === "text");
}

const active = isSymbolSearchOn() || isSearchText()
```

It also seems like SymbolSearch and SearchText are two different things that are at odds with eachother. You mentioned elsewhere that you have problems with the flow typing, maybe we need a new function, which is "searchTextBtn" with a new variable that has its own flow type? That might simplify the code.